### PR TITLE
named integration ids to contain integration name hint

### PR DESCRIPTION
**Pull request by itayd**
it is useful for integration ids that are built based on names to include a hint for the name inside them. this change tries to embed the original name in the id itself.

for example, for http:

`int_3kth00httpf1201a7ed83f7cd5`

this allows for easier debugging when need to know what integration this id belongs to.

### CODE REVIEW

File: sdk/sdktypes/integration_id.go [hunk 1]
```diff
@@ -22,8 +23,33 @@ func StrictParseIntegrationID(s string) (IntegrationID, error) { return Strict(P
 
 func IsIntegrationID(s string) bool { return IsIDOf[integrationIDTraits](s) }
 
-func IntegrationIDFromName(name string) IntegrationID {
-	txt := fmt.Sprintf("%s_%026x", integrationIDKind, kittehs.HashString64(name))
+const integrationIDFromNamePrefix = "3kth"
```
Comment (daabr) [inline sdk/sdktypes/integration_id.go:26]: I still think this prefix isn't useful. It's obvious that this kind of ID is special and looks different from regular UUIDs, and while you see "❤️🐱", I see "3000th".

Perosnally, I think it's better to use 0's to separate the name and the hash, i.e. `"<prefix> _ 00 %06s 00 %016s"`

But it's a stylistic choice, so I won't block on it.

File: sdk/sdktypes/integration_id.go [hunk 2]
```diff
@@ -22,8 +23,33 @@ func StrictParseIntegrationID(s string) (IntegrationID, error) { return Strict(P
...
+			return 'v'
+		default:
+			return r
+		}
+	}, name)
+
+	idName = fmt.Sprintf("%06s", idName)
+	if len(idName) > 6 {
+		idName = idName[:6]
+	}
```
Comment (daabr) [inline sdk/sdktypes/integration_id.go:45]: Limit the name's length before the character mapping to avoid wasted work.

**Review by daabr — APPROVED** [review]
Please limit the integration name's length to 6 bytes before the character mapping into a valid base32 string, to avoid wasted work.

File: sdk/sdktypes/integration_id.go (hunk 1 above)
Comment (itayd) [reply sdk/sdktypes/integration_id.go:26]: you can't put 0 in the beginning of an id, uuid complains. so i chose 3. i needed some more chars to make it a bit more distinguishable than a random 0000.

File: sdk/sdktypes/integration_id.go (hunk 2 above)
Comment (itayd) [reply sdk/sdktypes/integration_id.go:45]: done

**Pull request merged**